File: samples/hello.py
#!/usr/bin/env python3

import os.path, sys
from fityk import Fityk

class GaussianFitter(Fityk):
    def __init__(self, filename):
        Fityk.__init__(self)
        if not os.path.isfile(filename):
            raise ValueError("File `%s' not found." % filename)
        self.filename = filename
        self.execute("@0 < '%s'" % filename)
        print("Data info:", self.get_info("data", 0))

    def run(self):
        self.execute("guess Gaussian")
        gauss = self.all_functions()[-1] # the last function (just created)
        print("initial Gaussian center: %g" % gauss.get_param_value("center"))
        print("Fitting %s ..." % self.filename)
        self.execute("fit")
        print("WSSR=", self.get_wssr())
        print("Gaussian center: %g" % gauss.get_param_value("center"))

    def save_session(self, filename):
        self.execute("info state >'%s'" % filename)

f = Fityk()
print(f.get_info("version"))
print("ln(2) =", f.calculate_expr("ln(2)"))
del f

g = GaussianFitter(os.path.join(os.path.dirname(sys.argv[0]), "nacl01.dat"))
g.run()
g.save_session("tmp_save.fit")

File: samples/hello.rb
#!/usr/bin/env ruby

# Example of using libfityk from ruby.
# To run this example, you need fityk extension for Ruby: after compiling
# fityk go to src/, run "make ruby" and copy swig/fityk.so.

require 'fityk'

class GaussianFitter < Fityk::Fityk
    def initialize(filename)
        super()
        raise "File `#{filename}' not found." unless File::exists? filename
        @filename = filename
        execute("@0 < '#{filename}'")
        puts "Data info: " + get_info("data", 0)
    end

    def run()
        execute("guess %gauss = Gaussian")
        puts "Fitting #{@filename}..."
        execute("fit")
        puts "WSSR=%g" % get_wssr()
        puts "Gaussian center: %.5g" % calculate_expr("%gauss.center")
    end

    def save_session(filename)
        execute("info state >'%s'" % filename)
    end
end

f = Fityk::Fityk.new
puts f.get_info("version")
puts "ln(2) = %.9f" % f.calculate_expr("ln(2)")
f = nil

gauss = GaussianFitter.new("nacl01.dat")
gauss.run()
gauss.save_session("tmp_save.fit")

